generator.go

```diff
@@ -163,7 +174,7 @@ func (g *Generator) GoCode(observedValue *ObservedValue) ([]byte, error) {
 		fmt.Fprintf(buffer, "// %s\n", g.packageComment)
 	}
 	fmt.Fprintf(buffer, "package %s\n", g.packageName)
-	imports := make(map[string]bool)
+	imports := g.imports
```

Comment:
Using `g.imports` here introduces a subtle bug. Since the call to `g.GoType` on the next line may mutate `imports`, it will actually mutate the `g.imports` map. This means that multiple calls to `g.GoCode` on the same `Generator` struct will all use and modify the same map, causing later results to include imports from earlier results.

The fix here is to make a copy of `g.imports` and pass that to `g.GoType`.